src/plugins.rb

```diff
@@ -118,8 +118,19 @@ def enable!
       Plugins.enable!(name)
     end
 
-    # XXX Extract class for loading plugin questions?
     def configure_questions
+      ConfigureQuestionsBuilder.build(self)
+    end
+  end
+
+  ConfigureQuestionsBuilder = Struct.new(:plugin) do
+    private_class_method :new
+
+    def self.build(plugin)
+      new(plugin).build
+    end
+
+    def build
```

Comment:
I think having the internal build method is useful:

a. it's descriptive of what we're doing, rather than being a generic name;

b. in general it's expected that `SomeClass .new` will give an instance of `SomeClass`, not some other object, but in this case we want to return a hash, and not an instance of `ConfigureQuestionsBuilder`; the instance of this only exists while `ConfigureQuestionsBuilder.build` is executing to encapsulate the methods and state involved in this process. Therefore I think this change would be surprising.

Reply:
Ok good point, it can remain as is then.